civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -2170,6 +2177,117 @@ function civihr_employee_portal_manager_approval_pick_days_callback($ajax, $acti
 }
 
 /**
+ * Function that processes approvals and rejections made by manager from the absence approval widget
+ *
+ * @param int $ajax
+ *   boolean value, 1 if it is an ajax call
+ * @param string $action
+ *   should always be "quick_approval"
+ * @param int $activity_id
+ *   Activity ID that identifies the absence to be processed
+ * @param int $status
+ *   ID of status that should be set to main absence and each absence per day (2: Accepted, 9: Rejected)
+ * @param string $type
+ *   Staus type that is to be set, either "accept" or "reject"
+ */
+function civihr_employee_portal_manager_approval_quick_approval($ajax, $action, $activity_id, $status, $type) {
+  // Civi init
+  civicrm_initialize();
+
+  try {
+    // Load main activity leave
+    $main_activity = civicrm_api3('Activity', 'get', ['sequential' => 1, 'id' => $activity_id])['values'][0];
+
+    // Get activities for each day of leave
+    $search_params = ['source_record_id' => $activity_id, 'activity_type_id' => $main_activity['activity_type_id']];
+    $absence_details = civicrm_api3('Activity', 'get', $search_params);
+
+    // Set activities' status for each day of leave
+    foreach ($absence_details['values'] as $current_absence) {
+      civicrm_api3('Activity', 'setvalue', ['id' => $current_absence['id'], 'field' => 'status_id', 'value' => $status]);
+    }
+
+    // Set main activity's status
+    civicrm_api3('Activity', 'setvalue', ['id' => $activity_id, 'field' => 'status_id', 'value' => $status]);
+
+    // Invoke rules event
+    _invoke_rules_on_absence_decision($main_activity, $type, $absence_details['values']);
+
+    $result = ['status' => 'success'];
+  } catch (CiviCRM_API3_Exception $e) {
+    $errorMessage = $e->getMessage();
+    $errorCode = $e->getErrorCode();
+    $errorData = $e->getExtraParams();
+
+    $result = [
+      'status' => 'error',
+      'error' => $errorMessage,
+      'error_code' => $errorCode,
+      'error_data' => $errorData,
+    ];
+  }
+
+  // Issue response
+  print drupal_json_output($result);
+  drupal_exit();
+}
+
+/**
+ * Helper function that builds required parameters to pass to rules_invoke_event on decision (accept/reject) by manager
+ *
+ * @param $main_activity
+ * @param $type
+ * @param $absence_details
+ */
+function _invoke_rules_on_absence_decision($main_activity, $type, $absence_details) {
```

Comment:
The purpose of this function is to call "approve_all_post_event" and "reject_all_post_event" rules respectively, so the e-mail is sent to all interested parties on each event.  However, I found building required parameters for the rule took a lot of code, so I thought it would be cleaner to refactor all that code into this function.